lightspark-sdk/src/commonMain/kotlin/com/lightspark/sdk/LightsparkCoroutinesClient.kt

```diff
@@ -255,20 +270,30 @@ class LightsparkCoroutinesClient private constructor(
      * @param nodeId The ID of the node for which to create the invoice.
      * @param amountMsats The amount of the invoice in milli-satoshis.
      * @param metadata The LNURL metadata payload field from the initial payreq response. This will be hashed and
-     * present in the h-tag (SHA256 purpose of payment) of the resulting Bolt 11 invoice.
+     *      present in the h-tag (SHA256 purpose of payment) of the resulting Bolt 11 invoice.
      * @param expirySecs The number of seconds until the invoice expires. Defaults to 1 day.
+     * @param signingPrivateKey The receiver's signing private key. Used to hash the receiver identifier.
+     * @param receiverIdentifier Optional identifier of the receiver. If provided, this will be hashed
+     *      and used for anonymized analysis.
      */
     suspend fun createUmaInvoice(
         nodeId: String,
         amountMsats: Long,
         metadata: String,
         expirySecs: Int? = null,
+        signingPrivateKey: ByteArray? = null,
+        receiverIdentifier: String? = null,
     ): Invoice {
         requireValidAuth()
 
         val md = MessageDigest.getInstance("SHA-256")
         val digest = md.digest(metadata.toByteArray())
         val metadataHash = digest.fold(StringBuilder()) { sb, it -> sb.append("%02x".format(it)) }.toString()
+        val receiverHash = if (receiverIdentifier != null && signingPrivateKey != null) {
```

Comment:
I wonder if we should throw an error if you pass receiverIdentifier, but not the signing private key.